cassandra_backup_service.py

```diff
@@ -1875,15 +1875,18 @@ def full_backup(self, columnfamily=None, thread_limit=4):
         self.cassandra.nodetool_snapshot(snapshot_name, columnfamily)
 
         if self.backup_repo:
-            self.manifest_manager.download_manifests(self.cassandra.host_id)
-            self.manifest_manager.update_snapshot_manifests(snapshot_name, columnfamily)
-            self.manifest_manager.upload_manifests(self.cassandra.host_id)
-
-            self.backup_repo.upload_snapshot(self.cassandra.host_id, self.cassandra.data_file_directories,
-                                             snapshot_name, thread_limit)
-
-            logging.info('Clearing snapshot {0} data'.format(snapshot_name))
-            self.cassandra.nodetool_clearsnapshot(snapshot_name)
+            try:
+                self.manifest_manager.download_manifests(self.cassandra.host_id)
+                self.manifest_manager.update_snapshot_manifests(snapshot_name, columnfamily)
+                self.manifest_manager.upload_manifests(self.cassandra.host_id)
+
+                self.backup_repo.upload_snapshot(self.cassandra.host_id, self.cassandra.data_file_directories,
+                                                 snapshot_name, thread_limit)
+            except Exception as exception:
+                logging.warning('Exception when uploading during full_backup: {0}'.format(exception))
+                logging.info('Clearing snapshot {0} data'.format(snapshot_name))
+                self.cassandra.nodetool_clearsnapshot(snapshot_name)
```

Comment:
Now you are only clearing when there is an exception!

Reply:
Fixed in this commit: https://github.com/onzra/cassandra_backup_service/pull/38/commits/ed221c0f1b235ff6367e526085610adf22c650e9

Python will raise the exception after the `finally`:
> If an exception occurs in any of the clauses and is not handled, the exception is temporarily saved. The finally clause is executed. If there is a saved exception it is re-raised at the end of the finally clause.